diff --git a/core/cas-server-core-authentication/src/main/java/org/apereo/cas/config/CasCoreAuthenticationConfiguration.java b/core/cas-server-core-authentication/src/main/java/org/apereo/cas/config/CasCoreAuthenticationConfiguration.java
index 56c71f6df1..c2ffa13638 100644
--- a/core/cas-server-core-authentication/src/main/java/org/apereo/cas/config/CasCoreAuthenticationConfiguration.java
+++ b/core/cas-server-core-authentication/src/main/java/org/apereo/cas/config/CasCoreAuthenticationConfiguration.java
@@ -121,8 +121,12 @@ public class CasCoreAuthenticationConfiguration {
             AnnotationAwareOrderComparator.sortIfNecessary(sortedConfigurers);
 
             sortedConfigurers.forEach(Unchecked.consumer(c -> {
-                LOGGER.trace("Configuring authentication execution plan [{}]", c.getName());
-                c.configureAuthenticationExecutionPlan(plan);
+		try {
+                	LOGGER.trace("Configuring authentication execution plan [{}]", c.getName());
+                	c.configureAuthenticationExecutionPlan(plan);
+		} catch(Throwable th) {
+        	       	LOGGER.error("Authentication execution plan [{}] configuration failed", c.getName(), th);
+	        }
             }));
             return plan;
         }
diff --git a/gradle.properties b/gradle.properties
index c460b8165e..23f0ca0ecd 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -2,7 +2,7 @@
 # Platform metadata for releases, POM generation, etc.
 ##################################################
 group=org.apereo.cas
-version=6.5.9-SNAPSHOT
+version=6.5.5
 projectUrl=https://www.apereo.org/cas
 projectInceptionYear=2004
 projectScmUrl=scm:git@github.com:apereo/cas.git
diff --git a/support/cas-server-support-ldap/src/main/java/org/apereo/cas/config/LdapAuthenticationConfiguration.java b/support/cas-server-support-ldap/src/main/java/org/apereo/cas/config/LdapAuthenticationConfiguration.java
index 1e1b4c3a61..7e869232f8 100644
--- a/support/cas-server-support-ldap/src/main/java/org/apereo/cas/config/LdapAuthenticationConfiguration.java
+++ b/support/cas-server-support-ldap/src/main/java/org/apereo/cas/config/LdapAuthenticationConfiguration.java
@@ -74,9 +74,13 @@ public class LdapAuthenticationConfiguration {
                 }
                 return true;
             }).forEach(l -> {
-                val handler = LdapUtils.createLdapAuthenticationHandler(l, applicationContext, servicesManager, ldapPrincipalFactory);
-                handler.setState(l.getState());
-                handlers.add(handler);
+		try {
+                	val handler = LdapUtils.createLdapAuthenticationHandler(l, applicationContext, servicesManager, ldapPrincipalFactory);
+                	handler.setState(l.getState());
+                	handlers.add(handler);
+		} catch(Throwable th) {
+                        LOGGER.error("Failure to initialize LDAP authentication handler for [{}]", ldapPrincipalFactory, th);
+                }
             });
             ldapAuthenticationHandlerSetFactoryBean.getObject().addAll(handlers);
             return BeanContainer.of(handlers);
